### PR TITLE
Index channel logs in elasticsearch

### PR DESCRIPTION
**Pull request by D-Ne0**
A very basic working version of indexing channel logs in elasticsearch. Let me know what changes are required in this pull request.

```
sudo yum install elasticsearch
sudo service elasticsearch start
npm install elasticsearch
```

To test if elasticsearch is running, open: <a href="http://localhost:9200">http://localhost:9200</a>

To see all indexed documents

```
curl -XGET 'localhost:9200/channel_logs/_search?pretty=true'
```

### CODE REVIEW

**Comment by rtnpro** [comment]
@D-Ne0 looks good :+1: . Also, please add some basic code to start pushing channel logs to ES.

**Comment by D-Ne0** [comment]
@rtnpro The code in this pull request does the same. Do you want me to push old channel logs in mongo db too to ES? If yes, it's a one time task so should I write a script?

**Comment by rtnpro** [comment]
@D-Ne0 Please add some settings variable to decide whether to use ES or not.

**Comment by D-Ne0** [comment]
Done

File: waartaa/server/lib/02_irc.js [hunk 1]
```diff
@@ -214,6 +214,17 @@ ChannelLogsManager = function () {
     function _insert (log) {
         ChannelLogs.insert(log, function (err, id) {});
         OldChannelLogs.insert(log, function (err, id) {});
+        if(CONFIG.ENABLE_ELASTIC_SEARCH) {
```
Comment (rtnpro) [inline waartaa/server/lib/02_irc.js]: why no space after `if`?

File: waartaa/server/lib/02_irc.js [hunk 2]
```diff
@@ -214,6 +214,17 @@ ChannelLogsManager = function () {
...
+          var es_client = elasticsearch.Client({
+              host: 'localhost:9200'
```
Comment (rtnpro) [inline waartaa/server/lib/02_irc.js]: keep `host` configurable, in `CONFIG`

File: waartaa/server/lib/02_irc.js [hunk 3]
```diff
@@ -214,6 +214,17 @@ ChannelLogsManager = function () {
     function _insert (log) {
         ChannelLogs.insert(log, function (err, id) {});
         OldChannelLogs.insert(log, function (err, id) {});
+        if(CONFIG.ENABLE_ELASTIC_SEARCH) {
+          var es_client = elasticsearch.Client({
```
Comment (rtnpro) [inline waartaa/server/lib/02_irc.js]: Let's use horntail notation, which is conventional in JS, e.g., `esClient`

**Comment by rtnpro** [comment]
@D-Ne0 

Rebase your code on top of latest master. Does https://gist.github.com/D-Ne0/10a479fd4b6c1c0f4c93 need to go in the ES server?

**Comment by D-Ne0** [comment]
Hey, I have rebased my code and tested the changes. It's working fine on my machine.

Not exactly sure what you meant by gist going into the ES server. As I have mentioned in comment we basically have to run this after ES is installed: https://gist.github.com/D-Ne0/10a479fd4b6c1c0f4c93#comment-1234856

**Comment by rtnpro** [comment]
@D-Ne0 Rebase this as well.

**Comment by rtnpro** [comment]
Closing this as all changes merged to `bookmarking` branch.

**Pull request closed without merging**